pkgs/development/compilers/zig/generic.nix

```
@@ -1,91 +1,140 @@
-, libxml2
-, zlib
-, coreutils
-, callPackage
-, ...
-}:
-
-args:
-
-stdenv.mkDerivation (finalAttrs: {
-  pname = "zig";
-
-  src = fetchFromGitHub {
-    owner = "ziglang";
-    repo = "zig";
-    rev = finalAttrs.version;
-    inherit (args) hash;
-  };
-
-  nativeBuildInputs = [
-    cmake
-    llvmPackages.llvm.dev
-  ];
-
-  buildInputs = [
-    libxml2
-    zlib
-  ] ++ (with llvmPackages; [
-    libclang
-    lld
-    llvm
-  ]);
-
-  # On Darwin, Zig calls std.zig.system.darwin.macos.detect during the build,
-  # which parses /System/Library/CoreServices/SystemVersion.plist and
-  # /System/Library/CoreServices/.SystemVersionPlatform.plist to determine the
-  # OS version. This causes the build to fail during stage 3 with
-  # OSVersionDetectionFail when the sandbox is enabled.
-  __impureHostDeps = lib.optionals stdenv.isDarwin [
-    "/System/Library/CoreServices/.SystemVersionPlatform.plist"
-    "/System/Library/CoreServices/SystemVersion.plist"
-  ];
-
-  env.ZIG_GLOBAL_CACHE_DIR = "$TMPDIR/zig-cache";
-
-  # Zig's build looks at /usr/bin/env to find dynamic linking info. This doesn't
-  # work in Nix's sandbox. Use env from our coreutils instead.
-  postPatch = if lib.versionAtLeast args.version "0.12" then ''
-    substituteInPlace lib/std/zig/system.zig \
-      --replace "/usr/bin/env" "${coreutils}/bin/env"
-  '' else ''
-    substituteInPlace lib/std/zig/system/NativeTargetInfo.zig \
-      --replace "/usr/bin/env" "${coreutils}/bin/env"
-  '';
-
-  doInstallCheck = true;
-  installCheckPhase = ''
-    runHook preInstallCheck
-
-    $out/bin/zig test --cache-dir "$TMPDIR/zig-test-cache" -I $src/test $src/test/behavior.zig
-
-    runHook postInstallCheck
-  '';
-
-  passthru = {
-    hook = callPackage ./hook.nix {
-      zig = finalAttrs.finalPackage;
-    };
-    cc = callPackage ./cc.nix {
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+  llvmPackages,
+  libxml2,
+  zlib,
+  coreutils,
+  callPackage,
+  version,
+  hash,
+  patches ? [ ],
+}@args:
+
+stdenv.mkDerivation (
+  finalAttrs:
+  let
+    common = callPackage ./common.nix {
+      inherit (finalAttrs) pname version;
       zig = finalAttrs.finalPackage;
     };
-    stdenv = callPackage ./stdenv.nix {
-      zig = finalAttrs.finalPackage;
+  in
+  {
+    pname = "zig";
+    inherit version;
+
+    src = fetchFromGitHub {
+      owner = "ziglang";
+      repo = "zig";
+      rev = finalAttrs.version;
+      inherit hash;
     };
-  };
-
-  meta = {
-    description = "General-purpose programming language and toolchain for maintaining robust, optimal, and reusable software";
-    homepage = "https://ziglang.org/";
-    changelog = "https://ziglang.org/download/${finalAttrs.version}/release-notes.html";
-    license = lib.licenses.mit;
-    maintainers = with lib.maintainers; [ andrewrk ] ++ lib.teams.zig.members;
-    mainProgram = "zig";
-    platforms = lib.platforms.unix;
-  };
-} // removeAttrs args [ "hash" ])
+
+    patches =
+      args.patches or [ ]
```

```suggestion
      args.patches
```
that default is already in the args

---

Done

---

Looks like this breaks eval and causes this error:
```
       error: attribute 'patches' missing

       at /ofborg/checkout/1/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-4/pkgs/development/compilers/zig/generic.nix:36:7:

           35|     patches =
           36|       args.patches
             |       ^
           37|       ++ lib.optionals (lib.versions.majorMinor finalAttrs.version == "0.9") [
```